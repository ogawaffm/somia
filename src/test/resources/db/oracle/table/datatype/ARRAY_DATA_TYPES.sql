BEGIN
    EXECUTE IMMEDIATE 'DROP TABLE TEST.ARRAY_DATA_TYPES';
EXCEPTION
    WHEN OTHERS THEN
        IF SQLCODE != -942 THEN
            RAISE;
        END IF;
END;
/
-- drop TEST.VARRAY_5_VARRAY_5_VARCHAR2_50_TYPE which is based on TEST.VARRAY_5_VARCHAR2_50_TYPE
BEGIN
    EXECUTE IMMEDIATE 'DROP TYPE TEST.VARRAY_5_VARRAY_5_VARCHAR2_50_TYPE';
EXCEPTION
    WHEN OTHERS THEN
        IF SQLCODE != -4043 THEN -- Fehlercode -4043 = ORA-04043: Object does not exist
            RAISE; -- Fehler erneut ausl√∂sen, wenn es ein anderer Fehler ist
        END IF;
END;
/
CREATE OR REPLACE TYPE TEST.VARRAY_5_VARCHAR2_50_TYPE AS VARRAY(5) OF VARCHAR2(50);
/
CREATE OR REPLACE TYPE TEST.VARRAY_5_NUMBER_2_1_TYPE AS VARRAY(5) OF NUMBER(2, 1);
/
CREATE OR REPLACE TYPE TEST.VARRAY_5_TIMESTAMP_3_TYPE AS VARRAY(5) OF TIMESTAMP(3);
/
CREATE OR REPLACE TYPE TEST.VARRAY_5_INTERVAL_DAY_3_TO_SECOND_1_TYPE AS VARRAY(5) OF INTERVAL DAY(3) TO SECOND(1);
/
CREATE OR REPLACE TYPE TEST.VARRAY_5_SDO_GEOMETRY_TYPE AS VARRAY(5) OF SDO_GEOMETRY;
/
CREATE OR REPLACE TYPE TEST.VARRAY_5_ANYTYPE_TYPE AS VARRAY(5) OF ANYTYPE;
/
CREATE OR REPLACE TYPE TEST.VARRAY_5_BFILE_TYPE AS VARRAY(5) OF BFILE;
/
CREATE OR REPLACE TYPE TEST.VARRAY_5_VARRAY_5_VARCHAR2_50_TYPE AS VARRAY(5) OF TEST.VARRAY_5_VARCHAR2_50_TYPE;

-- No VARAY for XMLTYPE, ROWID and UROWID
/
CREATE TABLE TEST.ARRAY_DATA_TYPES
(

    VARRAY_5_VARCHAR2_50_COL                TEST.VARRAY_5_VARCHAR2_50_TYPE,
    VARRAY_5_NUMBER_2_1_COL                 TEST.VARRAY_5_NUMBER_2_1_TYPE,
    VARRAY_5_TIMESTAMP_3_COL                TEST.VARRAY_5_TIMESTAMP_3_TYPE,
    VARRAY_5_INTERVAL_DAY_3_TO_SECOND_1_COL TEST.VARRAY_5_INTERVAL_DAY_3_TO_SECOND_1_TYPE,
    VARRAY_5_SDO_GEOMETRY_COL               TEST.VARRAY_5_SDO_GEOMETRY_TYPE,
    VARRAY_5_ANYTYPE_COL                    TEST.VARRAY_5_ANYTYPE_TYPE,
    VARRAY_5_BFILE_COL                      TEST.VARRAY_5_BFILE_TYPE,
    VARRAY_5_VARRAY_5_VARCHAR2_50_TYPE      TEST.VARRAY_5_VARRAY_5_VARCHAR2_50_TYPE


)