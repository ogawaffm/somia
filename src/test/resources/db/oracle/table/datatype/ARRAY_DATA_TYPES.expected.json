{
  "name": "ARRAY_DATA_TYPES",
  "catalogName": "",
  "schemaName": "TEST",
  "typeName": "TABLE",
  "comment": "",
  "columns": [
    {
      "name": "VARRAY_5_VARCHAR2_50_COL",
      "type": {
        "name": "VARRAY_5_VARCHAR2_50_TYPE",
        "typeNumber": 2003,
        "className": "java.sql.Array",
        "precision": 272,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "VARRAY_5_NUMBER_2_1_COL",
      "type": {
        "name": "VARRAY_5_NUMBER_2_1_TYPE",
        "typeNumber": 2003,
        "className": "java.sql.Array",
        "precision": 132,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "VARRAY_5_TIMESTAMP_3_COL",
      "type": {
        "name": "VARRAY_5_TIMESTAMP_3_TYPE",
        "typeNumber": 2003,
        "className": "java.sql.Array",
        "precision": 77,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "VARRAY_5_INTERVAL_DAY_3_TO_SECOND_1_COL",
      "type": {
        "name": "VARRAY_5_INTERVAL_DAY_3_TO_SECOND_1_TYPE",
        "typeNumber": 2003,
        "className": "java.sql.Array",
        "precision": 77,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "VARRAY_5_SDO_GEOMETRY_COL",
      "type": {
        "name": "VARRAY_5_SDO_GEOMETRY_TYPE",
        "typeNumber": 2003,
        "className": "java.sql.Array",
        "precision": 3752,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "VARRAY_5_ANYTYPE_COL",
      "type": {
        "name": "VARRAY_5_ANYTYPE_TYPE",
        "typeNumber": 2003,
        "className": "java.sql.Array",
        "precision": 3752,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "VARRAY_5_BFILE_COL",
      "type": {
        "name": "VARRAY_5_BFILE_TYPE",
        "typeNumber": 2003,
        "className": "java.sql.Array",
        "precision": 2692,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    },
    {
      "name": "VARRAY_5_VARRAY_5_VARCHAR2_50_TYPE",
      "type": {
        "name": "VARRAY_5_VARRAY_5_VARCHAR2_50_TYPE",
        "typeNumber": 2003,
        "className": "java.sql.Array",
        "precision": 1402,
        "scale": 0,
        "isNullable": true,
        "isAutoIncrement": false,
        "numericPrecisionRadix": 10,
        "isCaseSensitive": false,
        "isSearchable": true,
        "isSigned": false,
        "isCurrency": false,
        "columnDisplaySize": 0,
        "baseType": null
      },
      "comment": "",
      "collation": "",
      "defaultDefinition": ""
    }
  ],
  "indexes": [
    {
      "name": "",
      "columns": [],
      "isUnique": true,
      "comment": "",
      "qualifierName": "",
      "isClustered": false,
      "isHashed": false,
      "isStatistic": true,
      "whereClause": "",
      "cardinality": 0,
      "pages": 0
    }
  ],
  "collation": "",
  "primaryKey": null,
  "bestRowIdentifier": null
}